ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -70,20 +71,38 @@ impl<'a> TraceBuilder<'a> {
             if inst.is_ptr_add() {
                 let op = inst.operand(0);
                 // unwrap safe: we know the AOT code was produced by ykllvm.
-                if trace_input
-                    .unwrap()
+                let ti_op = trace_input.unwrap();
+                if ti_op
                     .to_instr(self.aot_mod)
                     .ptr_eq(op.to_instr(self.aot_mod))
                 {
-                    // Found a trace input.
+                    // We found a trace input. Now we emit a `LoadArg` instruction into the trace.
+                    // This assigns the input a local variable that other instructions can then
+                    // use.
+                    //
+                    // Note: This code assumes that the `PtrAdd` instructions in the AOT IR were
+                    // emitted sequentially.
+                    //
                     // unwrap safe: we know the AOT code was produced by ykllvm.
                     let inp = last_store.unwrap().operand(0);
+                    let trace_input_struct_ty = ti_op.type_(self.aot_mod).as_struct();
+                    let aot_field_ty_idx = trace_input_struct_ty.field_type_idx(trace_input_idx);
+                    // FIXME: assumes the field is byte-aligned. If it isn't, field_byte_off() will
+                    // crash.
+                    let aot_field_off = trace_input_struct_ty.field_byte_off(trace_input_idx);
+                    let jit_field_ty_idx = self.handle_type(aot_field_ty_idx)?;
                     input.insert(0, inp.to_instr(self.aot_mod));
-                    let load_arg = jit_ir::LoadArgInstruction::new().into();
-                    self.local_map
-                        .insert(inp.to_instr_id(), self.next_instr_id()?);
-                    self.jit_mod.push(load_arg);
-                    self.first_ti_idx = inst_idx;
+                    if let Ok(u32_off) = u32::try_from(aot_field_off) {
```

Comment:
This is more conventionally a `match` rather than `if let`.

Reply:
Since there are only two possibilities, and we don't consume the error, they should be equivalent. Up to you, but hardly worth the trouble, is it?

Reply:
It is a minor thing, but it means that I almost misread the code, because it's the less common idiom.

Reply:
Changed to a match in cdb0bab1a96dc4dfde4cc5648c9cbe9796718abb